### PR TITLE
CORDA-3224 JDK11 built and published artifacts to include classifier.

### PR DESCRIPTION
**Pull request by josecoll**
Introduction of a POM classifier to differentiate JDK11 built and published artefacts from Java 8 equivalents.

This PR is dependent on small changes to corda gradle plugins: https://github.com/corda/corda-gradle-plugins/pull/251

### CODE REVIEW

File: tools/shell-cli/build.gradle [hunk 1]
```diff
@@ -35,12 +35,11 @@ assemble.dependsOn buildShellCli
 
 artifacts {
     publish shadowJar {
-        classifier = ""
+        archiveClassifier = jdkClassifier
     }
```
Comment (chrisr3) [inline tools/shell-cli/build.gradle:39]: The `shadow` plugin doesn't always "Do The Right Thing" with classifiers - I'd check the jars explicitlyto ensure the artifacts are correct.

File: tools/network-builder/build.gradle [hunk 2]
```diff
@@ -73,7 +73,7 @@ processResources {
 
 shadowJar {
     baseName = 'network-builder'
-    classifier = null
+    archiveClassifier = jdkClassifier
```
Comment (chrisr3) [inline tools/network-builder/build.gradle:76]: Check these artifacts - the `shadow` plugin's handling of `archiveClassifier` could be buggy.

File: tools/explorer/capsule/build.gradle [hunk 3]
```diff
@@ -17,7 +17,7 @@ capsule {
 
 task buildExplorerJAR(type: FatCapsule, dependsOn: project(':tools:explorer').tasks.jar) {
     applicationClass 'net.corda.explorer.Main'
-    archiveName "node-explorer-${corda_release_version}.jar"
+    archiveName jdkClassifier == null ? "node-explorer-${corda_release_version}.jar" : "node-explorer-${corda_release_version}-${jdkClassifier}.jar"
```
Comment (chrisr3) [inline tools/explorer/capsule/build.gradle]: I'm sure that `archiveName` could be replaced by `archiveBaseName`, `archiveVersion` and `archiveClassifier`. [See here](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.bundling.Jar.html).

File: tools/shell-cli/build.gradle [hunk 4]
```diff
@@ -35,12 +35,11 @@ assemble.dependsOn buildShellCli
...
 }
 
 jar {
-    classifier "ignore"
```
Comment (chrisr3) [inline tools/shell-cli/build.gradle:43]: This was here to ensure that the `jar` artifact would have a different name ("fail safe"). Perhaps replace `classifier` by `archiveClassifier` if you want to remove a warning?

**Review by chrisr3 — COMMENTED** [review]
I'm worried that the `shadow` plugin may be misbehaving because it has known bugs with classifiers. It is _definitely_ worth checking.

And I suspect that the `capsule` builds don't need to make quite such a manual effort with `archiveName` either.

File: tools/shell-cli/build.gradle (hunk 1 above)
Comment (josecoll) [reply tools/shell-cli/build.gradle:39]: The jar looks correct to me:

```
Joses-MBP-2:corda-jdk11 josecoll$ ./gradlew tools:shell-cli:clean tools:shell-cli:buildShellCli

> Configure project :
Java version: 11
Java source compatibility: 1.8
Java target compatibility: 11
Quasar version: 0.8.0
Quasar classifier: null

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 13s
40 actionable tasks: 4 executed, 36 up-to-date
Joses-MBP-2:corda-jdk11 josecoll$ !ls
ls -l tools/shell-cli/build/libs/
total 124016
-rw-r--r--  1 josecoll  staff  62529699 25 Sep 12:48 shell-cli-4.3-SNAPSHOT-jdk11.jar
```

File: tools/network-builder/build.gradle (hunk 2 above)
Comment (josecoll) [reply tools/network-builder/build.gradle:76]: Ditto:

```
Joses-MBP-2:corda-jdk11 josecoll$ ./gradlew tools:network-builder:clean tools:network-builder:buildNetworkBuilder

> Configure project :
Java version: 11
Java source compatibility: 1.8
Java target compatibility: 11
Quasar version: 0.8.0
Quasar classifier: null

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 27s
49 actionable tasks: 6 executed, 43 up-to-date
Joses-MBP-2:corda-jdk11 josecoll$ ls -l tools/network-builder/build/libs/
total 231376
-rw-r--r--  1 josecoll  staff  112655804 25 Sep 12:50 network-builder-jdk11.jar
```

File: tools/explorer/capsule/build.gradle (hunk 3 above)
Comment (josecoll) [reply tools/explorer/capsule/build.gradle]: `archiveClassifier = jdkClassifier` does not seem to take effect.

Comment (josecoll) [reply tools/explorer/capsule/build.gradle]: Resolved with additional gradle magic dust.

**Comment by josecoll** [comment]
@chris back to you.

File: build.gradle [hunk 5]
```diff
@@ -25,10 +25,13 @@ buildscript {
     if (JavaVersion.current() == JavaVersion.VERSION_11) {
         ext.quasar_version = constants.getProperty("quasarVersion11")
         ext.quasar_classifier = constants.getProperty("quasarClassifier11")
+        // specify a classifier to be included in all built (and published) Corda JAR artifacts
+        ext.jdkClassifier='jdk11'
```
Comment (chrisr3) [inline build.gradle]: Eh? So the classifier for JDK11 is hard-coded but the one for JDK8 is read from `constants.properties`?

File: testing/testserver/testcapsule/build.gradle [hunk 6]
```diff
@@ -26,7 +26,7 @@ capsule {
 
 task buildWebserverJar(type: FatCapsule, dependsOn: project(':node').tasks.jar) {
     applicationClass 'net.corda.webserver.WebServer'
-    archiveName "corda-testserver-${corda_release_version}.jar"
+    archiveName jdkClassifier == null ? "corda-testserver-${corda_release_version}.jar" : "corda-testserver-${corda_release_version}-${jdkClassifier}.jar"
```
Comment (chrisr3) [inline testing/testserver/testcapsule/build.gradle]: The Node and WebServer are both Capsules, just like Node Explorer.

**Review by chrisr3 — COMMENTED** [review]
Does DemoBench still build/run correctly with JDK8?

File: testing/testserver/testcapsule/build.gradle (hunk 6 above)
Comment (josecoll) [reply testing/testserver/testcapsule/build.gradle]: And?

**Comment by josecoll** [comment]
@chrisr3 please add **all** your comments in a single PR review  and not incrementally after every time I update the PR (with new comments).

File: testing/testserver/testcapsule/build.gradle (hunk 6 above)
Comment (josecoll) [reply testing/testserver/testcapsule/build.gradle]: Fixed.

**Comment by josecoll** [comment]
@chrisr3 back to you.

**Comment by josecoll** [comment]
> Does DemoBench still build/run correctly with JDK8?

**DemoBench does work** ...wWith exception of an existing problem you have already raised in relation to starting the Web Server (https://github.com/JetBrains/jdk8u/issues/6) and which is totally independent of this PR.

**Review by chrisr3 — APPROVED** [review]
LGTM

**Pull request merged**